### PR TITLE
Remember deployment details with diagram

### LINKED ISSUE #1066: Remember deployment details with diagram
__Is your feature request related to a problem? Please describe.__

The deployment feature is meant to be used repeatedly (you are likely to update assets once published). Unfortunately as a user I do need to type the details (deployment name, tenant id) over and over again. 

__Describe the solution you'd like__

* [ ] The modeler should remember the entered details
* [ ] Re-publishing is as simple as opening the dialog and pressing [ENTER] or the deploy button

__Describe alternatives you've considered__

An interesting question is in which scope the details should be remembered:

* per diagram (persisted in XML) 
* per tab (stored transiently as long as the diagram tab is opened)
* in the _modeler workspace_ or _to be invented project scope_

__Solution Sketch__

If we go for the diagram.file.path prefix in the global configuration approach:

API to retrieve information per diagram:

```
config.setForDiagram(diagram, key, value);
config.getForDiagram(diagram, key, defaultValue);
```

__Considerations__

* Save As should update per diagram config entries.

---

Related to https://github.com/camunda/camunda-modeler/issues/1067, child of https://github.com/camunda/camunda-modeler/issues/1439

Related to [Support Issue](https://app.camunda.com/jira/browse/SUPPORT-5416)

### PR DESCRIPTION
**Pull request by barmac**
Closes #1066

### CODE REVIEW

File: client/src/plugins/deployment-tool/__tests__/DeploymentToolSpec.js [hunk 1]
```diff
@@ -84,6 +72,50 @@ describe('<DeploymentTool>', () => {
...
+
+      const { handleClose } = wrapper.state('modalState');
+
+      handleClose(details);
+
+      await nextTick();
+
+      // then
+      expect(config.getForFile).to.have.been.calledOnce;
+      expect(config.setForFile).to.have.been.calledOnce;
```
Comment (pinussilvestrus) [inline client/src/plugins/deployment-tool/__tests__/DeploymentToolSpec.js]: Maybe let's also test for which file and with which details it got called?

File: client/src/plugins/deployment-tool/DeploymentDetailsModal.js [hunk 2]
```diff
@@ -125,14 +149,27 @@ export default class DeploymentDetailsModal extends React.PureComponent {
     return this.lazilyCheckConnection(values);
   }
 
-  getEndpointConfigFields(authType) {
+  getEndpointConfigHint(values, errors) {
```
Comment (pinussilvestrus) [inline client/src/plugins/deployment-tool/DeploymentDetailsModal.js]: This seems to be not quite covered by tests right now, maybe we can add some test cases for this?

File: client/src/plugins/deployment-tool/__tests__/DeploymentToolSpec.js (hunk 1 above)
Comment (barmac) [reply client/src/plugins/deployment-tool/__tests__/DeploymentToolSpec.js]: OK

File: client/src/plugins/deployment-tool/DeploymentDetailsModal.js (hunk 2 above)
Comment (barmac) [reply client/src/plugins/deployment-tool/DeploymentDetailsModal.js]: So now we test whether the hint is displayed.

**Comment by barmac** [comment]
@pinussilvestrus ready for round two

**Pull request merged**